packages/contracts-bedrock/test/L2/L1Block.t.sol

```diff
@@ -93,3 +61,71 @@ contract L1BlockTest is CommonTest {
 }
+
+contract L1BlockEcotone_Test is L1BlockTest {
+    /// @dev Tests that setL1BlockValuesEcotone updates the values appropriately.
+    function testFuzz_setL1BlockValuesEcotone_succeeds(
+        uint32 basefeeScalar,
+        uint32 blobBasefeeScalar,
+        uint64 sequenceNumber,
+        uint64 timestamp,
+        uint64 number,
+        uint256 basefee,
+        uint256 blobBasefee,
+        bytes32 hash,
+        bytes32 batcherHash
+    )
+        external
+    {
+        bytes memory functionCallDataPacked = Encoding.encodeSetL1BlockValuesEcotone(
+            basefeeScalar, blobBasefeeScalar, sequenceNumber, timestamp, number, basefee, blobBasefee, hash, batcherHash
+        );
+
+        vm.prank(depositor);
+        (bool success,) = address(l1Block).call(functionCallDataPacked);
+        require(success, "Function call failed");
+
+        assertEq(l1Block.basefeeScalar(), basefeeScalar);
+        assertEq(l1Block.blobBasefeeScalar(), blobBasefeeScalar);
+        assertEq(l1Block.sequenceNumber(), sequenceNumber);
+        assertEq(l1Block.timestamp(), timestamp);
+        assertEq(l1Block.number(), number);
+        assertEq(l1Block.basefee(), basefee);
+        assertEq(l1Block.blobBasefee(), blobBasefee);
+        assertEq(l1Block.hash(), hash);
+        assertEq(l1Block.batcherHash(), batcherHash);
+
+        // ensure we didn't accidentally pollute the 64 bits of the number & timestamp slot that
+        // should be empty
+        bytes32 numberTimestampSlot = vm.load(address(l1Block), bytes32(uint256(0)));
+		bytes32 mask = hex"FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF00000000000000000000000000000000";
+        assertEq(0, numberTimestampSlot & mask);
+    }
+
+    /// @dev Tests that `setL1BlockValuesEcotone` succeeds if sender address is the depositor
+    function test_setL1BlockValuesEcotone_isDepositor_succeeds() external {
+        bytes memory functionCallDataPacked = Encoding.encodeSetL1BlockValuesEcotone(
+            type(uint32).max, type(uint32).max, type(uint64).max, type(uint64).max, type(uint64).max, type(uint256).max,
+            type(uint256).max ,bytes32(type(uint256).max), bytes32(type(uint256).max)
+        );
+
+        vm.prank(depositor);
+        (bool success,) = address(l1Block).call(functionCallDataPacked);
+        require(success, "function call failed");
+    }
+
+    /// @dev Tests that `setL1BlockValuesEcotone` fails if sender address is not the depositor 
+    function test_setL1BlockValuesEcotone_notDepositor_fails() external {
+        bytes memory functionCallDataPacked = Encoding.encodeSetL1BlockValuesEcotone(
+            type(uint32).max, type(uint32).max, type(uint64).max, type(uint64).max, type(uint64).max, type(uint256).max,
+            type(uint256).max ,bytes32(type(uint256).max), bytes32(type(uint256).max)
+        );
+
+        (bool success, bytes memory data) = address(l1Block).call(functionCallDataPacked);
+        require(!success, "function call should have failed but didn't");
```

Comment:
nit: single `'` will mess up the color formatting, would prefer if you remove it

Reply:
removed